packages/compass-query-bar/src/components/query-bar.spec.tsx

```diff
@@ -18,6 +35,61 @@ const exportToLanguageButtonId = 'query-bar-open-export-to-language-button';
 const queryHistoryButtonId = 'query-history-button';
 const queryHistoryComponentTestId = 'query-history';
 
+const getFilterInputEditorView = () => {
+  const filterInput = screen.getByTestId(
+    'query-bar-option-filter-input'
+  ) as any;
+  return filterInput._cm as EditorView;
+};
+
+const initiateFilterBarWithText = async (text: string) => {
+  const editorView = getFilterInputEditorView();
+  await waitUntilEditorIsReady(editorView);
+
+  editorView.dispatch({
+    changes: { from: 0, to: editorView.state.doc.length, insert: text },
+  });
+
+  await waitUntilEditorIsReady(editorView);
+};
+
+const getFilterInputContent = async () => {
+  const editorView = getFilterInputEditorView();
+  await waitUntilEditorIsReady(editorView);
+
+  return editorView.state.doc.sliceString(0) ?? '';
+};
```

Comment:
We expose `setCodemirrorEditorValue` from the editor package for testing purposes, to avoid having a direct dependency on codemirror here, we might add a similarly implemented `getCodemirrorEditorValue` here